### PR TITLE
Ansari frontend version 1.0.0

### PR DESCRIPTION
**Pull request by EndeavorPal**
Ansari Frontend Release 1.0.0

### CODE REVIEW

File: src/interfaces/message.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface MessageModel {
+  role: string
```
Comment (waleedkadous) [inline src/interfaces/message.ts:2]: Could we align this with the other places where roles are defined? In fact there's an almost identical class in the project.

File: src/reportWebVitals.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { ReportHandler } from 'web-vitals'
+
+const reportWebVitals = (onPerfEntry?: ReportHandler) => {
+  if (onPerfEntry && onPerfEntry instanceof Function) {
+    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
+      getCLS(onPerfEntry)
```
Comment (waleedkadous) [inline src/reportWebVitals.ts:6]: Can you add a comment to explain what these abbreviations stand for or at least where I can look up what they mean?

**Review by waleedkadous — APPROVED** [review]

File: src/components/home/Message.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import { logo, messageLoader, user, warningCircle } from 'assets'
+import { ErrorMessage } from 'components/ui'
+import React from 'react'
+import { Trans } from 'react-i18next'
+
+interface MessageProps {
+  type: 'receiver' | 'sender'
```
[the remark this answers is not in the document]
Comment (EndeavorPal) [reply src/components/home/Message.tsx:7]: @waleedkadous We use this interface to identify the UI only. Roles are handled in the Role enum.

File: src/components/home/SuggestionMessages.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import { challenge, chat, praying } from 'assets'
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { useDispatch } from 'react-redux'
+import { addMessage } from 'store/message-slice'
+
+const SuggestionMessages: React.FC = () => {
```
[the remark this answers is not in the document]
Comment (EndeavorPal) [reply src/components/home/SuggestionMessages.tsx:7]: @waleedkadous Could you please specify the text that should be displayed to the user? and when this message should appear to the user.

File: src/constant/index.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export enum Role {
+  ASSISTANT = 'assistant',
+  USER = 'user',
+}
```
[the remark this answers is not in the document]
Comment (EndeavorPal) [reply src/constant/index.ts:4]: @waleedkadous could you please explain more what the functionality behind each role

- **user**
- **assistant**
- function
- system

File: src/i18n/index.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import i18n from 'i18next'
+import { initReactI18next } from 'react-i18next'
+import ar from './locales/ar.json'
+import bs from './locales/bs.json'
+import en from './locales/en.json'
+import id from './locales/id.json'
```
[the remark this answers is not in the document]
Comment (EndeavorPal) [reply src/i18n/index.ts:6]: Will be available with the next release 1.0.1

File: src/components/ui/LanguageSelectorPopUp.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    },
+  },
+}
+
+type LanguageObject = {
+  [key: string]: {
+    nativeName: string
+  }
+}
+const languages: LanguageObject = {
```
[the remark this answers is not in the document]
Comment (EndeavorPal) [reply src/components/ui/LanguageSelectorPopUp.tsx:18]: Will be available with the next release 1.0.1

File: src/components/home/Message.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (waleedkadous) [reply src/components/home/Message.tsx:7]: Got it. Makes sense.

File: src/components/home/SuggestionMessages.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (waleedkadous) [reply src/components/home/SuggestionMessages.tsx:7]: @EndeavorPal  I had something like this in mind: take the text from the Information on the side, and at the start page have it look like this. 

image

File: src/components/ui/LanguageSelectorPopUp.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (waleedkadous) [reply src/components/ui/LanguageSelectorPopUp.tsx:18]: Excellent!

File: src/constant/index.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (waleedkadous) [reply src/constant/index.ts:4]: I gave this some thought. It's OK. Let's jut stick with user and assistant. 

I can keep function and system messages to the backend for now. System messages define the personality of the LLM, function give results.

File: src/i18n/index.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (waleedkadous) [reply src/i18n/index.ts:6]: Excellent!

File: src/reportWebVitals.ts (hunk 2 above)
Comment (EndeavorPal) [reply src/reportWebVitals.ts:6]: @waleedkadous  This a measuring performance tool, you can have a look at it for more details https://create-react-app.dev/docs/measuring-performance/

**Pull request merged**